### DISCUSSION TITLE
Why do max-width and min-width classes not provide the same values as width classes?

### DISCUSSION THREAD

**Discussion by o-t-w** [tailwindlabs/tailwindcss#9681] [Help] [answered]
I have defined my spacing scale in the config file. I want to use classes like `min-w-16`. Why does Tailwind allow classes like `w-16` but not equivalent classes for `min-width` and `max-width`? I found this very surprising. 

Even min-width and max-width classes are not consistent in Tailwind. max-width decides not to use the numeric class names but instead uses `sm`, `md`, `lg` etc. I really do not like this inconsistency. 
What is the best way to change my config file so that I can use my spacing scale in a consistent way in width min and max width? I realize that in the config file I can edit `theme.minWidth` and `theme.width` and `theme.maxWidth` but I don't want to have to repeat the scale 4 times in the same file!

**Comment by adamwathan** [comment, marked as answer]
The values were chosen in the v1 days when we had no JIT engine and had to be very mindful of file size, so the values are heavily curated for each utility based on common use cases we identified in our own work. Could probably unify them more now just hasn’t happened because there’s been other more important things to work on.

> but I don't want to have to repeat the scale 4 times in the same file!

It’s a JavaScript file, create a variable. Alternatively add all of those values to your `spacing` config and update the other keys to all inherit the spacing values.
